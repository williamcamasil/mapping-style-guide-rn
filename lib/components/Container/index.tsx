import React from 'react';
import { View, ViewProps, ViewStyle } from 'react-native';

import { useViewStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';

export type ContainerPropsType = ViewProps & {
  theme: AppThemeType;
  padding?: ViewStyle['padding'];
  paddingVertical?: ViewStyle['padding'];
  paddingHorizontal?: ViewStyle['padding'];
  paddingLeft?: ViewStyle['padding'];
  paddingRight?: ViewStyle['padding'];
  paddingTop?: ViewStyle['padding'];
  paddingBottom?: ViewStyle['padding'];
  fillParent?: boolean;
  horizontal?: boolean;
  centered?: boolean;
  backgroundColor?: ViewStyle['backgroundColor'];
  blockActions?: boolean;
  justifyContent?: ViewStyle['justifyContent'];
  alignItems?: ViewStyle['alignItems'];
};

const Container: React.FC<ContainerPropsType> = ({
  children,
  style,
  theme,
  padding,
  paddingVertical,
  paddingHorizontal,
  paddingLeft,
  paddingRight,
  paddingTop,
  paddingBottom,
  fillParent,
  horizontal,
  centered,
  backgroundColor,
  blockActions,
  justifyContent,
  alignItems,
  ...others
}) => {
  const containerStyles = useViewStyles(() => [
    {
      flex: fillParent ? 1 : undefined,
      flexDirection: horizontal ? 'row' : undefined,
      justifyContent: justifyContent ?? (centered && horizontal ? 'center' : undefined),
      alignItems: alignItems ?? (centered && !horizontal ? 'center' : undefined),
      backgroundColor: backgroundColor || theme.colors.neutralWhite,
      padding,
      paddingVertical,
      paddingHorizontal,
      paddingLeft,
      paddingRight,
      paddingTop,
      paddingBottom,
    },
    style,
  ], [
    backgroundColor,
    centered,
    fillParent,
    horizontal,
    padding,
    paddingBottom,
    paddingHorizontal,
    paddingLeft,
    paddingRight,
    paddingTop,
    paddingVertical,
    style,
    theme.colors.neutralWhite,
    justifyContent,
    alignItems,
  ]);

  return (
    <View pointerEvents={blockActions ? 'none' : 'auto'} style={containerStyles} {...others}>
      {children}
    </View>
  );
};

Container.defaultProps = {
  fillParent: true,
};

export default withTheme(Container);
