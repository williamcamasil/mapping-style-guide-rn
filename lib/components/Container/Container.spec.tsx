import React from 'react';
import { View } from 'react-native';

import renderer from 'react-test-renderer';

import Container from '.';

describe('Container snapshot', () => {
  it('fill parent', () => {
    const tree = renderer
      .create(
        <Container fillParent>
          <View />
        </Container>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('dont fill parent', () => {
    const tree = renderer
      .create(
        <Container fillParent={false}>
          <View />
        </Container>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('centered', () => {
    const tree = renderer
      .create(
        <Container centered>
          <View />
        </Container>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('horizontal', () => {
    const tree = renderer
      .create(
        <Container horizontal>
          <View />
        </Container>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('horizontal centered', () => {
    const tree = renderer
      .create(
        <Container horizontal centered>
          <View />
        </Container>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('blockActions', () => {
    const tree = renderer
      .create(
        <Container blockActions>
          <View />
        </Container>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('justifyContent and alignItems', () => {
    const tree = renderer
      .create(
        <Container justifyContent="center" alignItems="flex-end">
          <View />
        </Container>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
