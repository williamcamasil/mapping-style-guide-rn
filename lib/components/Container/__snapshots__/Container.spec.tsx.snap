// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container snapshot blockActions 1`] = `
<View
  pointerEvents="none"
  style={
    {
      "alignItems": undefined,
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "flexDirection": undefined,
      "justifyContent": undefined,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "paddingVertical": undefined,
    }
  }
>
  <View />
</View>
`;

exports[`Container snapshot centered 1`] = `
<View
  pointerEvents="auto"
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "flexDirection": undefined,
      "justifyContent": undefined,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "paddingVertical": undefined,
    }
  }
>
  <View />
</View>
`;

exports[`Container snapshot dont fill parent 1`] = `
<View
  pointerEvents="auto"
  style={
    {
      "alignItems": undefined,
      "backgroundColor": "#FFFFFF",
      "flex": undefined,
      "flexDirection": undefined,
      "justifyContent": undefined,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "paddingVertical": undefined,
    }
  }
>
  <View />
</View>
`;

exports[`Container snapshot fill parent 1`] = `
<View
  pointerEvents="auto"
  style={
    {
      "alignItems": undefined,
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "flexDirection": undefined,
      "justifyContent": undefined,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "paddingVertical": undefined,
    }
  }
>
  <View />
</View>
`;

exports[`Container snapshot horizontal 1`] = `
<View
  pointerEvents="auto"
  style={
    {
      "alignItems": undefined,
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": undefined,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "paddingVertical": undefined,
    }
  }
>
  <View />
</View>
`;

exports[`Container snapshot horizontal centered 1`] = `
<View
  pointerEvents="auto"
  style={
    {
      "alignItems": undefined,
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "paddingVertical": undefined,
    }
  }
>
  <View />
</View>
`;

exports[`Container snapshot justifyContent and alignItems 1`] = `
<View
  pointerEvents="auto"
  style={
    {
      "alignItems": "flex-end",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "flexDirection": undefined,
      "justifyContent": "center",
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "paddingVertical": undefined,
    }
  }
>
  <View />
</View>
`;
